rusk-wallet/src/main.rs

```diff
@@ -211,16 +205,17 @@ struct Rusk {
 }
 
 /// Connect to rusk services via TCP
-async fn rusk_tcp(rusk_addr: String, prov_addr: String) -> Result<Rusk, Error> {
+async fn rusk_tcp(rusk_addr: &str, prov_addr: &str) -> Result<Rusk, Error> {
     Ok(Rusk {
-        network: NetworkClient::connect(rusk_addr.clone()).await?,
-        state: StateClient::connect(rusk_addr).await?,
-        prover: ProverClient::connect(prov_addr).await?,
+        network: NetworkClient::connect(rusk_addr.to_string()).await?,
+        state: StateClient::connect(rusk_addr.to_string()).await?,
+        prover: ProverClient::connect(prov_addr.to_string()).await?,
     })
 }
 
 /// Connect to rusk via UDS (Unix domain sockets)
-async fn rusk_uds(socket_path: String) -> Result<Rusk, Error> {
+async fn rusk_uds(socket_path: &str) -> Result<Rusk, Error> {
+    let socket_path = socket_path.to_string();
```

Comment:
We are converting `socket_path` to string, and later on we convert it again to string.

[UnixStream::connect](https://doc.rust-lang.org/std/os/unix/net/struct.UnixStream.html#method.connect) accepts anything that can be a `Path`. Looks like we are making an unnecessary complication here

Reply:
This gave me a lot of trouble in the past. I really don't know how to solve it. The borrow checker complains with anything I throw at it. I'd appreciate help with this one @vlopes11